File: Leetcode第20题.cpp
class Solution {
public:
    bool isValid(string s) {
        if(s.size()%2){return false;}
        stack<char> stk;
        for(char c:s){
            switch (c)
            {
                case '(':
                stk.push(')');
                break;
                case '[':
                stk.push(']');
                break;
                case '{':
                stk.push('}');
                break;
                default:
                if(stk.size()==0||stk.top()!=c){
                    return false;
                }else{
                    stk.pop();
                }
            }
        }
        if(stk.size()==0){
            return true;
        }else{
            return false;
        }
    }
};

File: Leetcode第20题.py
class Solution(object):
    def isValid(self, s):
        """
        :type s: str
        :rtype: bool
        """
        if(len(s)&1  
        or s.count('(')!=s.count(')') 
        or s.count('[')!=s.count(']') 
        or s.count('{')!=s.count('}')) :
            return False
        stack=["#"]
        for c in s:
            if c in '([{':
                stack.append(c)
            else:
                if stack.pop()+c not in ("()","[]","{}"):
                    return False
        return True
